test/views/behaviours/ViewBehaviours.scala

```diff
@@ -23,60 +23,82 @@ trait ViewBehaviours extends ViewSpecBase {
+      "have the correct banner title" in {
+        val doc         = asDocument(viewViaApply)
+        val bannerTitle = doc.getElementsByClass("hmrc-header__service-name")
+        bannerTitle.text mustBe messages(app)("service.name")
+      }
 
-        "display the correct browser title" in {
-          val doc = asDocument(view)
-          assertEqualsMessage(doc, "title", s"$messageKeyPrefix.title$suffix")
-        }
+      "display the correct guidance" in {
+        val doc = asDocument(viewViaApply)
+        for (key <- expectedGuidanceKeys)
+          assertContainsText(doc, messages(app)(s"$messageKeyPrefix.$key"))
+      }
 
-        "display the correct page title" in {
-          val doc = asDocument(view)
-          assertPageTitleEqualsMessage(doc, s"$messageKeyPrefix.heading$suffix", messageHeadingArgs: _*)
+      "contain a timeout dialog" in {
+        val timeoutElm = asDocument(viewViaApply).select("meta[name=\"hmrc-timeout-dialog\"]")
+        if (expectTimeoutDialog) {
+          assert(timeoutElm.size == 1)
+        } else {
+          assert(timeoutElm.size == 0)
         }
+      }
+    }
 
-        "display the correct guidance" in {
-          val doc = asDocument(view)
-          for (key <- expectedGuidanceKeys)
-            assertContainsText(doc, messages(applicationBuilder().build())(s"$messageKeyPrefix.$key"))
-        }
+  protected def pageByMethod(
+    view: HtmlFormat.Appendable,
+    messageKeyPrefix: String,
+    messageKeySuffix: String,
+    messageHeadingArgs: Any*
+  )(byMethod: String = "", isError: Boolean = false): Unit = {
 
-        "contain a timeout dialog" in {
-          val timeoutElm = asDocument(view).select("meta[name=\"hmrc-timeout-dialog\"]")
-          if (expectTimeoutDialog) {
-            assert(timeoutElm.size == 1)
-          } else {
-            assert(timeoutElm.size == 0)
-          }
-        }
-      }
+    val suffix = if (messageKeySuffix.isEmpty) "" else s".$messageKeySuffix"
+
+    val renderMethod = if (byMethod.nonEmpty) {
+      s"when rendered - using $byMethod"
+    } else {
+      "when rendered"
     }
 
-  def pageWithExpectedMessages(view: HtmlFormat.Appendable, checks: Seq[(String, String)]): Unit =
-    checks.foreach { case (cssSelector, message) =>
-      s"element with cssSelector '$cssSelector'" - {
+    s"$renderMethod" - {
 
-        s"have message '$message'" in {
-          val doc  = asDocument(view)
-          val elem = doc.select(cssSelector)
-          elem.first.text() mustBe message
-        }
+      "display the correct browser title" in {
+        val doc = asDocument(view)
+        assertEqualsMessage(doc, "title", s"$messageKeyPrefix.title$suffix", messageHeadingArgs: _*)(isError)
+      }
+
+      "display the correct page heading (h1)" in {
+        val doc = asDocument(view)
+        assertPageHeadingEqualsMessage(doc, s"$messageKeyPrefix.heading$suffix", messageHeadingArgs: _*)
       }
     }
+  }
+
+  def pageByMethodWithAssertions(
```

Comment:
Also, please remove By in the naming of pageByMethodWithAssertions and pageByMethod so it is `pageMethod...` as this was done for the parameter so all `By` references in this trait are removed, thanks